openupgrade_scripts/scripts/hr/15.0.1.1/post-migration.py

```diff
@@ -0,0 +1,68 @@
+from openupgradelib import openupgrade
+
+
+def compute_work_location_id(env):
+    env.cr.execute("SELECT * FROM hr_employee WHERE work_location IS NOT NULL")
+    all_employees = env.cr.dictfetchall()
+
+    work_location_dict = {}
+    for employee in all_employees:
+        if employee["work_location"] not in work_location_dict.keys():
+            address_id = employee.get("address_id", False)
+            if not address_id:
+                address_id = (
+                    env["res.partner"]
+                    .create(
+                        {
+                            "name": employee["work_location"],
+                            "company_type": "person",
+                            "type": "private",
+                        }
+                    )
+                    .id
+                )
+            work_location = env["hr.work.location"].create(
+                {
+                    "name": employee["work_location"],
+                    "company_id": employee["company_id"],
+                    "address_id": address_id,
+                }
+            )
+            work_location_dict[employee["work_location"]] = work_location
+        env["hr.employee"].browse(employee["id"]).work_location_id = work_location_dict[
+            employee["work_location"]
+        ]
+
+
+def compute_departure_reason(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE hr_employee
+            SET departure_reason_id = CASE departure_reason
+                    WHEN 'fired' THEN %s
+                    WHEN 'resigned' THEN %s
+                    WHEN 'retired' THEN %s
+                END
+            WHERE departure_reason IS NOT NULL
```

Comment:
@daonamutc1 tránh trường hợp module khác bổ sung thêm tùy chọn khác
```suggestion
            WHERE departure_reason IN ('fired', 'resigned', 'retired')
```